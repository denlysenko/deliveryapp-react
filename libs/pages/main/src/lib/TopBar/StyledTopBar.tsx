import styled from 'styled-components';

export const StyledTopBar = styled.div`
  .topbar {
    position: fixed;
    width: 100%;
    left: 0;
    top: 0;
    color: #757575;
    height: 70px;
    padding: 12px 15px;
    border-bottom: solid 1px #dee4e9;
    z-index: 100;
    background-color: white;
    background-image: linear-gradient(to right, white, white);

    a {
      color: #757575;
    }

    &-logo {
      .fa {
        font-size: 44px;
        vertical-align: middle;
        color: #3984b8;
      }

      @media (max-width: 640px) {
        width: 144px;
        display: block;
        margin: 0 auto;
        text-align: center;
      }
    }

    &-search {
      vertical-align: middle;
      border: 0 none;
      background-color: transparent;
      color: #757575;
      transition: color 0.3s;

      @media (max-width: 640px) {
        display: none;
      }
    }

    &-badge {
      position: absolute;
      top: -8px;
      right: -6px;
      background-color: #3eb839;
      padding: 2px 8px;
      display: block;
      margin-top: -2px;
      color: white;
      border-radius: 2px;
    }

    .fa-search {
      font-size: 16px;
      vertical-align: middle;
      margin-left: 36px;

      @media (max-width: 640px) {
        display: none;
      }
    }
  }

  #menu-button {
    font-size: 24px;
    float: left;
    margin-left: 0;
    margin-top: -36px;
    @media (min-width: 641px) {
      display: none;
    }
  }

  #user-display {
    position: relative;
    float: right;
    text-align: right;
    margin-top: 2px;
    text-decoration: none;

    @media (max-width: 640px) {
      margin-top: -42px;
    }

    .username {
      transition: color 0.3s;
      vertical-align: middle;
      color: #757575;
      margin-right: 8px;

      @media (max-width: 640px) {
        display: none;
      }
    }

    img {
      vertical-align: middle;
      width: 40px;
      margin-right: 8px;
    }

    .fa {
      transition: color 0.3s;
      font-size: 16px;
      vertical-align: middle;
      color: #757575;
    }
  }

  .p-menu {
    width: 250px;

    .p-menuitem-link {
      font-weight: 300;
      color: #7d8286;
      padding: 10px 16px;

      &.ui-state-active {
        color: #3984b8;
        background-color: #f6f9fb;
      }

      &:hover {
        background-color: #f6f9fb;
      }
    }
  }
`;
