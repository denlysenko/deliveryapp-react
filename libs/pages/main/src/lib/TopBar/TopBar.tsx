import React, { useRef, useState } from 'react';
import { Link, useHistory } from 'react-router-dom';

import { InputText } from 'primereact/inputtext';
import { Menu } from 'primereact/menu';

import { isNil } from 'lodash-es';

import { useAuth, logout } from '@deliveryapp/data-access';

import { StyledTopBar } from './StyledTopBar';

interface TopBarProps {
  showMessages: () => void;
}

export const TopBar: React.FC<TopBarProps> = ({ showMessages }) => {
  const menuRef = useRef<Menu>(null);
  const history = useHistory();
  const [{ user }, dispatch] = useAuth();

  const [items] = useState([
    {
      label: 'Profile',
      icon: 'fa fa-user',
      command: () => {
        history.push('/profile');
      }
    },
    {
      label: 'Messages',
      icon: 'fa fa-envelope-o',
      command: () => {
        showMessages();
      }
    },
    {
      label: 'Logout',
      icon: 'fa fa-sign-out',
      command: () => {
        logout(dispatch, history);
      }
    }
  ]);

  const toggleUserMenu = (
    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
  ) => {
    event.preventDefault();
    !isNil(menuRef.current) && menuRef.current.toggle(event);
  };

  const toggleAppMenu = (
    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
  ) => {
    event.stopPropagation();
    event.preventDefault();
    const layout = document.querySelector('.layout');
    !isNil(layout) && layout.classList.toggle('layout-menu-active');
  };

  return (
    <StyledTopBar>
      <div className="topbar">
        <Link className="topbar-logo" to="/">
          <i className="fa fa-clock-o"></i>
        </Link>
        <a href="/" id="menu-button" onClick={toggleAppMenu}>
          <i className="fa fa-bars"></i>
        </a>
        <a
          id="user-display"
          data-testid="menu"
          href="/"
          onClick={toggleUserMenu}
        >
          {user?.firstName && (
            <span className="username">{user?.firstName}</span>
          )}
          {user?.lastName && <span className="username">{user?.lastName}</span>}
          {!user?.firstName && !user?.lastName && (
            <span className="username">{user?.email}</span>
          )}
          <img src="/assets/icons/avatar.svg" alt="avatar" />
          <span className="topbar-badge">0</span>
          <i className="fa fa-angle-down"></i>
        </a>
        <Menu model={items} popup={true} ref={menuRef} />
        <label htmlFor="topbar-search">
          <i className="fa fa-search topbar-search-icon"></i>
        </label>
        <InputText
          className="topbar-search"
          id="topbar-search"
          placeholder="Search..."
        />
      </div>
    </StyledTopBar>
  );
};
