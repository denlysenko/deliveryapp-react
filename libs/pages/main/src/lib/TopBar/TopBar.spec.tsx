import React from 'react';
import { render } from '@testing-library/react';

import TopBar from './TopBar';

describe(' TopBar', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<TopBar />);
    expect(baseElement).toBeTruthy();
  });
});
