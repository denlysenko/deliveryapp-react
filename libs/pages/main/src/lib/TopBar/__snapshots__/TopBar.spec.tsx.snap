// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-AykKC drFxTE"
    >
      <div
        class="topbar"
      >
        <a
          class="topbar-logo"
          href="/"
        >
          <i
            class="fa fa-clock-o"
          />
        </a>
        <a
          href="/"
          id="menu-button"
        >
          <i
            class="fa fa-bars"
          />
        </a>
        <a
          data-testid="menu"
          href="/"
          id="user-display"
        >
          <span
            class="username"
          >
            User
          </span>
          <span
            class="username"
          >
            Test
          </span>
          <img
            alt="avatar"
            src="/assets/icons/avatar.svg"
          />
          <span
            class="topbar-badge"
          >
            0
          </span>
          <i
            class="fa fa-angle-down"
          />
        </a>
        <div
          class="p-menu p-component p-menu-dynamic p-menu-overlay"
        >
          <ul
            class="p-menu-list p-reset"
            role="menu"
          >
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="fa fa-user p-menuitem-icon"
                />
                <span
                  class="p-menuitem-text"
                >
                  Profile
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="fa fa-envelope-o p-menuitem-icon"
                />
                <span
                  class="p-menuitem-text"
                >
                  Messages
                </span>
              </a>
            </li>
            <li
              class="p-menuitem"
              role="none"
            >
              <a
                class="p-menuitem-link"
                href="#"
                role="menuitem"
              >
                <span
                  class="fa fa-sign-out p-menuitem-icon"
                />
                <span
                  class="p-menuitem-text"
                >
                  Logout
                </span>
              </a>
            </li>
          </ul>
        </div>
        <label
          for="topbar-search"
        >
          <i
            class="fa fa-search topbar-search-icon"
          />
        </label>
        <input
          class="p-inputtext p-component topbar-search"
          id="topbar-search"
          placeholder="Search..."
        />
      </div>
    </div>
  </div>
</body>
`;
